apis/druid/v1alpha1/druid_types.go

```diff
@@ -266,17 +267,45 @@ type DeepStorageSpec struct {
 type DruidStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+	// Conditions are the latest available observations of a druid nodes's state.
+	Conditions           []DruidNodeConditions `json:"druidNodeConditions,omitempty"`
```

Comment:
As a user, I care about following things...

- Are all Druid node sts/deployment spec's up-to-date?
- Are all pods up-to-date or some sts/deployment is in middle of updates?
- Are there any errors?
- Is cluster overall up-to-date and healthy ?

current proposal doesn't help me understand "status of Druid cluster" but state transitions which is secondary .... e.g. say druid-operator successfully updated the sts spec but pods are stuck in some error never becoming ready ...  change proposed here wouldn't show that anything is wrong.

Reply:
very much valid points. I'd be refining the proposal.
thanks @himanshug